sql/analyzer/pushdown.go

```diff
@@ -223,7 +223,19 @@ func transformPushdownFilters(ctx *sql.Context, a *Analyzer, n sql.Node, scope *
 					return nil, err
 				}
 				return FixFieldIndexesForExpressions(ctx, a, n, scope)
-			case *plan.TableAlias, *plan.ResolvedTable, *plan.IndexedTableAccess, *plan.ValueDerivedTable:
+			case *plan.IndexedTableAccess:
+				table := getTable(node)
+				if table == nil {
+					return n, nil
+				}
+				if _, ok := table.(sql.IndexAddressableTable); ok {
+					lookup := indexes[table.Name()]
+					if lookup != nil {
+						filters.markFiltersHandled(lookup.expr)
```

Comment:
This is a roundabout way of doing this. You should mark the filters handled at the same time you apply the index to the table. See other usages of markFiltersHandled for examples.

Reply:
Correct me if I'm reading this wrong, but all other usages of `markFiltersHandled` are in `pushdownFiltersToTable` in the case statement below this one. We mark filters as handled at this point because the `filtersSet` was just constructed from the calling scope.